google-cloud-storage/src/test/java/com/google/cloud/storage/ITMultipartUploadHttpRequestManagerTest.java

```diff
@@ -376,4 +379,92 @@ public void sendCreateMultipartUploadRequest_withCustomTime() throws Exception {
           endpoint, request, httpStorageOptions);
     }
   }
+
+  @Test
+  public void sendListPartsRequest_success() throws Exception {
+    HttpRequestHandler handler =
+        req -> {
```

Comment:
We should perform some request validation to ensure the proper request is being sent to the server. something like only return ok when the bucket, key and uploadId match expected. otherwise 404 or 400 with a message.